abdiff/core/collate_ab_transforms.py

```diff
@@ -228,31 +324,43 @@ def validate_output(dataset_path: str) -> None:
         )
 
         # check if the table is empty
-        record_count = con.execute("SELECT COUNT(*) FROM collated").fetchone()[0]  # type: ignore[index]
+        record_count = fetch_single_value("SELECT COUNT(*) FROM collated")
         if record_count == 0:
-            raise OutputValidationError(  # noqa: TRY003
+            raise OutputValidationError(
                 "The collated dataset does not contain any records."
             )
 
         # check if any of the 'record_*' columns are empty
-        record_a_null_count = con.execute(
+        record_a_null_count = fetch_single_value(
             "SELECT COUNT(*) FROM collated WHERE record_a ISNULL"
-        ).fetchone()[
-            0
-        ]  # type: ignore[index]
-
-        record_b_null_count = con.execute(
+        )
+        record_b_null_count = fetch_single_value(
             "SELECT COUNT(*) FROM collated WHERE record_b ISNULL"
-        ).fetchone()[
-            0
-        ]  # type: ignore[index]
+        )
 
         if record_count in {record_a_null_count, record_b_null_count}:
-            raise OutputValidationError(  # noqa: TRY003
+            raise OutputValidationError(
                 "At least one or both record column(s) ['record_a', 'record_b'] "
                 "in the collated dataset are empty."
             )
 
+        # check that timdex_record_id column is unique
+        non_unique_count = fetch_single_value(
+            """
+            SELECT COUNT(*)
+            FROM (
+                SELECT timdex_record_id
+                FROM collated
+                GROUP BY timdex_record_id
+                HAVING COUNT(*) > 1
+            ) as duplicates;
+        """
+        )
+        if non_unique_count > 0:
+            raise OutputValidationError(
+                "The collated dataset contains duplicate 'timdex_record_id' records."
+            )
```

Comment:
Unclear if this is too much to ask (please push back if so), but it seems it would be helpful to know which `timdex_record_id` is duplicated

Reply:
I hear ya, but I would pushback, for reasons of scale.  If this were not working correctly, it's conceivable that 10's, 100's, thousands of records could be duplicated.  I would posit it's sufficient that if _any_ records are duplicated, something is intrinsically wrong with the deduplication logic; it's that that needs attention, and not a specific record.

Counter-point: we could show a sample like 10 records.  And then during debugging that work, you could look for those?   But... I suppose my preference would be to skip that for now, unless we have trouble with this in the future.

Reply:
Fine with me, the scale argument makes perfect sense! Agree there's no need to add unless we find it to be a problem